### PR TITLE
feature/issue 18

### PR DESCRIPTION
**Pull request by Mykytaaa**
Global Exception Handling

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -16,4 +16,21 @@ management:
   tracing:
     enabled: true
     sampling:
-      probability: 1.0
\ No newline at end of file
+      probability: 1.0
```
Comment (dobmax) [inline src/main/resources/application.yml:19]: please do rebase

File: src/main/java/com/mykytaaa/user/profile/userprofileservice/rest/dto/ApiResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.mykytaaa.user.profile.userprofileservice.rest.dto;
+
+import java.util.List;
+
+public record ApiResponseDto(int status, String message, List<String> details) {
```
Comment (maksymdobrynin) [inline src/main/java/com/mykytaaa/user/profile/userprofileservice/rest/dto/ApiResponseDto.java]:
```suggestion
public record ErrorDto(int status, String message, List<String> details) {
```

File: src/main/java/com/mykytaaa/user/profile/userprofileservice/rest/dto/ApiResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.mykytaaa.user.profile.userprofileservice.rest.dto;
```
Comment (dobmax) [inline src/main/java/com/mykytaaa/user/profile/userprofileservice/rest/dto/ApiResponseDto.java:1]:
```suggestion
package com.mykytaaa.user.profile.userprofileservice.exception;
```

File: src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java [hunk 4]
```diff
@@ -101,6 +104,36 @@ void shouldSuccessfullyCreateUser_withUserDetails(SoftAssertions softly) {
...
+        StepVerifier.create(webTestClient.post()
+                .uri(URL_USER_CONTROLLER)
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(testUserCreateRequestDto))
+                .exchange()
+                .expectStatus().isBadRequest()
+                .expectHeader().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+                .returnResult(ApiResponseDto.class)
+                .getResponseBody())
+                .assertNext(exception -> {
```
Comment (maksymdobrynin) [inline src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java]:
```suggestion
                .assertNext(response -> {
```

File: src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java [hunk 5]
```diff
@@ -101,6 +104,36 @@ void shouldSuccessfullyCreateUser_withUserDetails(SoftAssertions softly) {
...
+                    softly.assertThat(exception.status()).isEqualTo(400);
+                    softly.assertThat(exception.message()).containsOnlyOnce("Email should be valid");
+                    softly.assertThat(exception.message()).containsOnlyOnce("default message [firstName],50,1]; default message [size must be between 1 and 50]]");
+                    softly.assertThat(exception.message()).containsOnlyOnce("default message [lastName],50,1]; default message [size must be between 1 and 50]]");
```
Comment (maksymdobrynin) [inline src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java]: assert as list of messages

File: src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java [hunk 6]
```diff
@@ -136,6 +169,32 @@ void shouldSuccessfullyUpdateUser(SoftAssertions softly) {
...
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(BodyInserters.fromValue(userUpdateRequestDto))
+                        .exchange()
+                        .expectStatus().isBadRequest()
+                        .expectHeader().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+                        .returnResult(ApiResponseDto.class)
+                        .getResponseBody())
+                .assertNext(exception -> {
+                    softly.assertThat(exception.status()).isEqualTo(400);
+                    softly.assertThat(exception.message()).contains("NotNull.firstName", "NotNull.lastName");
```
Comment (maksymdobrynin) [inline src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java]: details missing

File: src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java [hunk 7]
```diff
@@ -101,6 +104,36 @@ void shouldSuccessfullyCreateUser_withUserDetails(SoftAssertions softly) {
...
+                .uri(URL_USER_CONTROLLER)
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(testUserCreateRequestDto))
+                .exchange()
+                .expectStatus().isBadRequest()
+                .expectHeader().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+                .returnResult(ApiResponseDto.class)
+                .getResponseBody())
+                .assertNext(exception -> {
+                    softly.assertThat(exception.status()).isEqualTo(400);
```
Comment (maksymdobrynin) [inline src/test/java/com/mykytaaa/user/profile/userprofileservice/service/rest/controller/UserControllerIntegrationTest.java]: details missing

**Review by maksymdobrynin — APPROVED** [review]

**Pull request merged**